### ISSUE TITLE
The problem of reading bytes via GetManyBytes() method.

### ISSUE THREAD

**Issue by binoopang** [williballenthin/python-idb#12]
Hi, Thank you for your nice project :)

When I try to read bytes on the IDB via `GetManyBytes` method, sometimes 
I got bytes from another location that i didn't intended. 

```
Name          Start    End      R W X D L Align  Base Type   Class AD es       ss       ds       fs       gs      
----          -----    ---      - - - - - -----  ---- ----   ----- -- --       --       --       --       --      
.init         080496AC 080496CF R . X . L dword  0001 public CODE  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.plt          080496D0 08049DE0 R . X . L para   0002 public CODE  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.plt.got      08049DE0 08049DE8 R . X . L qword  0003 public CODE  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.text         08049DF0 0805B634 R . X . L para   0004 public CODE  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.fini         0805B634 0805B648 R . X . L dword  0005 public CODE  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.rodata       0805B660 08060C14 R . . . L 32byte 0006 public CONST 32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.eh_frame_hdr 08060C14 08061430 R . . . L dword  0007 public CONST 32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.eh_frame     08061430 080664D8 R . . . L dword  0008 public CONST 32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.init_array   08067F00 08067F04 R W . . L dword  0009 public DATA  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.fini_array   08067F04 08067F08 R W . . L dword  000A public DATA  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.jcr          08067F08 08067F0C R W . . L dword  000B public DATA  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.got          08067FFC 08068000 R W . . L dword  000C public DATA  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.got.plt      08068000 080681CC R W . . L dword  000D public DATA  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.data         080681E0 08068344 R W . . L 32byte 000E public DATA  32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
.bss          08068380 08068FB8 R W . . L 64byte 000F public BSS   32 FFFFFFFF FFFFFFFF 000E     FFFFFFFF FFFFFFFF
extern        08068FB8 08069188 ? ? ? . L para   0010 public       32 FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF
```

With this segments, I've try to read bytes at 0x8049df0, which is first byte of .text section. The 
code is following

``` python
 v = api.idc.GetManyBytes(0x8049df0, 0x10)
    for i in range(0x10):
        print '%02x' % ord(v[i]),
```
And the result is following

```
53 83 ec 08 e8 5b 28 00 00 81 c3 4b e9 01 00 8b
```

The actual bytes at 0x8049df0 is following 

``` 
.text:08049DF0 8D 4C 24 04                             lea     ecx, [esp+4]
.text:08049DF4 83 E4 F0                                and     esp, 0FFFFFFF0h
.text:08049DF7 FF 71 FC                                push    dword ptr [ecx-4]
.text:08049DFA 55                                      push    ebp
.text:08049DFB 89 E5                                   mov     ebp, esp
.text:08049DFD 57                                      push    edi
```
You can see it is different. The bytes that returned by `GetManyBytes` is actually code of `_init_proc`, which lives in .init section. 

```
.init:080496AC 53                                      push    ebx             ; _init
.init:080496AD 83 EC 08                                sub     esp, 8
.init:080496B0 E8 5B 28 00 00                          call    sub_804BF10
.init:080496B5 81 C3 4B E9 01 00                       add     ebx, 1E94Bh
.init:080496BB 8B 83 FC FF FF FF                       mov     eax, ds:(__gmon_start___ptr - 8068000h)[ebx]
.init:080496C1 85 C0                                   test    eax, eax
.init:080496C3 74 05                                   jz      short loc_80496CA
.init:080496C5 E8 16 07 00 00                          call    __gmon_start__
```

I think the problem is the `offset` of segment descriptor. when I executed following code, 
I got the value of seg.offset `0`.

```
 seg = api.idc.idb.id1.get_segment(0x8049df0)
 print 'seg offset: 0x%x' % seg.offset
```

Since the offset of flag is calculated by `seg.offset + 4 * (0x8049df0 - 0x8049df0)`, 
the calculated flag offset is 0. That's why I got the bytes of _init_proc function instead of 
main function. 

I made a following link that I used to perform this experiment. This is just a IDB of /bin/ls on 
Ubuntu OS 32bit. And the executable binary `ls` is 32bit ELF binary. 

[ls.idb](http://150.95.132.43/ls.idb)

Thank you.

**Comment by williballenthin** [comment]
thanks for reporting this issue @binoopang! i appreciate the effort you went through to provide a thorough bug report *and* test case. i'll address this issue soon.

**Comment by williballenthin** [comment]
underlying issue appears to be that ID1 segments are not being parsed correctly. this was hidden in prior unit tests because the `.text` section happened to always be found first.

**Comment by williballenthin** [comment]
issue fixed in 6349e5fd57eb29741c6c80b7104208db61515a82

thanks again to @binoopang for reporting the issue!
